src/Workspaces/CSharp/Portable/Formatting/CSharpFormattingOptions.cs

```diff
@@ -1,148 +1,351 @@
+            }
+        }
 
         public static Option<bool> IndentBraces { get; } = new Option<bool>(nameof(CSharpFormattingOptions), nameof(IndentBraces), defaultValue: false,
-            storageLocations: new RoamingProfileStorageLocation("TextEditor.CSharp.Specific.OpenCloseBracesIndent"));
+            storageLocations: new OptionStorageLocation[] {
+                new EditorConfigStorageLocation("csharp_indent_braces"),
+                new RoamingProfileStorageLocation("TextEditor.CSharp.Specific.OpenCloseBracesIndent")});
 
         public static Option<bool> IndentBlock { get; } = new Option<bool>(nameof(CSharpFormattingOptions), nameof(IndentBlock), defaultValue: true,
-            storageLocations: new RoamingProfileStorageLocation("TextEditor.CSharp.Specific.IndentBlock"));
+            storageLocations: new OptionStorageLocation[] {
+                new EditorConfigStorageLocation("csharp_indent_block_contents"),
+                new RoamingProfileStorageLocation("TextEditor.CSharp.Specific.IndentBlock")});
 
         public static Option<bool> IndentSwitchSection { get; } = new Option<bool>(nameof(CSharpFormattingOptions), nameof(IndentSwitchSection), defaultValue: true,
-            storageLocations: new RoamingProfileStorageLocation("TextEditor.CSharp.Specific.IndentSwitchSection"));
+            storageLocations: new OptionStorageLocation[] {
+                new EditorConfigStorageLocation("csharp_indent_case_labels"),
+                new RoamingProfileStorageLocation("TextEditor.CSharp.Specific.IndentSwitchSection")});
 
         public static Option<bool> IndentSwitchCaseSection { get; } = new Option<bool>(nameof(CSharpFormattingOptions), nameof(IndentSwitchCaseSection), defaultValue: true,
-            storageLocations: new RoamingProfileStorageLocation("TextEditor.CSharp.Specific.IndentSwitchCaseSection"));
+            storageLocations: new OptionStorageLocation[] {
+                new EditorConfigStorageLocation("csharp_indent_case_contents"),
+                new RoamingProfileStorageLocation("TextEditor.CSharp.Specific.IndentSwitchCaseSection")});
 
         public static Option<LabelPositionOptions> LabelPositioning { get; } = new Option<LabelPositionOptions>(nameof(CSharpFormattingOptions), nameof(LabelPositioning), defaultValue: LabelPositionOptions.OneLess,
-            storageLocations: new RoamingProfileStorageLocation("TextEditor.CSharp.Specific.LabelPositioning"));
+            storageLocations: new OptionStorageLocation[] {
+                new EditorConfigStorageLocation("csharp_indent_label", ParseEditorConfigLabelPositioning),
+                new RoamingProfileStorageLocation("TextEditor.CSharp.Specific.LabelPositioning")});
+
+        private static object ParseEditorConfigLabelPositioning(string lableIndentationValue)
+        {
+            switch (lableIndentationValue)
+            {
+                case "left_most": return LabelPositionOptions.LeftMost;
```

Comment:
LabelPositionOptions kind of messed this up, but you can fix it for editorconfig!